livekit-agents/livekit/agents/llm/llm.py

```diff
@@ -1,36 +1,18 @@
 from __future__ import annotations
 
 import abc
-import enum
 from typing import Callable
 
 from attrs import define
 
+from .chat_context import ChatContext, ChatRole
 from .function_context import FunctionContext
 
 
-class ChatRole(enum.Enum):
-    SYSTEM = "system"
-    USER = "user"
-    ASSISTANT = "assistant"
-    TOOL = "tool"
-
-
-@define
-class ChatMessage:
-    role: ChatRole
-    text: str
-
-
-@define
-class ChatContext:
-    messages: list[ChatMessage] = []
```

Comment:
This was using a shared list for all ChatContext instances. Moved to a field with factory in chat_context.py